Study pull request

ArseniyKorotkov opened the pull request.

ArseniyKorotkov committed:
start try to write code

ArseniyKorotkov committed:
start try to write code

ArseniyKorotkov committed:
add auditor
fix addMissingExercise

ArseniyKorotkov committed:
add auditor
fix addMissingExercise

ArseniyKorotkov committed:
add auditor
fix addMissingExercise

ArseniyKorotkov committed:
without .idea

YlabCourse commented on pom.xml:
```diff
@@ -0,0 +1,41 @@
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <version>5.10.0</version>
```
версии зависимостей в properties

YlabCourse commented on src/main/java/by/y_lab/p1entity/Exercise.java:
```diff
@@ -0,0 +1,34 @@
+    }
+
+
+    @Override
```
у тебя же подключен ломбок используй его

YlabCourse commented on src/test/java/DiaryTest.java:
```diff
@@ -0,0 +1,64 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class DiaryTest {
```
юнит тесты тестируют каждый класс изолировано от остальных. Принято называть их [Имя класса]Test

YlabCourse commented on src/main/java/by/y_lab/p0util/Auditor.java:
```diff
@@ -0,0 +1,23 @@
+package by.y_lab.p0util;
```
порядковые номера из имени пакетов убрать и нижнее подчеркивание тоже

YlabCourse commented on src/main/java/by/y_lab/p0util/Auditor.java:
```diff
@@ -0,0 +1,23 @@
+import java.util.ArrayList;
+import java.util.List;
+
+
```
лишняя пустая строка

YlabCourse commented on src/main/java/by/y_lab/p0util/Auditor.java:
```diff
@@ -0,0 +1,23 @@
+import java.util.List;
+
+
+public class Auditor {
```
нет javadoc

YlabCourse commented on src/main/java/by/y_lab/p0util/Auditor.java:
```diff
@@ -0,0 +1,23 @@
+
+public class Auditor {
+
+    private static final List<String> auditList = new ArrayList<>();
```
аудит это не список строк. Задача аудит иметь возможность ответить на вопрос кто/когда/что делал. Как будем искать по строкам?

YlabCourse commented on src/main/java/by/y_lab/p0util/FormatDateTime.java:
```diff
@@ -0,0 +1,17 @@
+/**
+ * Форматтеры дат
+ */
+public class FormatDateTime {
```
финал класс и приватный конструктор добавить  для утилитного класса

ArseniyKorotkov committed:
Remake architecture in Diary:
- delete Diary.java
- methods rebase in DiaryDto
Add tests
Rename packages
Remake audit function

ArseniyKorotkov committed:
Merge branch 'main' into tryBranch
